cedar-policy-validator/src/cedar_schema/err.rs

```diff
@@ -102,26 +103,27 @@ lazy_static! {
 pub struct ParseError {
     /// Error generated by lalrpop
     err: OwnedRawParseError,
+    /// Source code
+    src: Arc<str>,
 }
 
-impl From<RawParseError<'_>> for ParseError {
-    fn from(err: RawParseError<'_>) -> Self {
+impl ParseError {
+    pub(crate) fn from_raw_parse_error(err: RawParseError<'_>, src: Arc<str>) -> Self {
         Self {
             err: err.map_token(|token| token.to_string()),
+            src,
         }
     }
-}
 
-impl From<RawErrorRecovery<'_>> for ParseError {
-    fn from(recovery: RawErrorRecovery<'_>) -> Self {
-        recovery.error.into()
+    pub(crate) fn from_raw_error_recovery(recovery: RawErrorRecovery<'_>, src: Arc<str>) -> Self {
+        Self::from_raw_parse_error(recovery.error, src)
     }
 }
 
 impl ParseError {
     /// Extract a primary source span locating the error.
     pub fn primary_source_span(&self) -> SourceSpan {
-        let Self { err } = self;
+        let Self { err, .. } = self;
         match err {
```

Comment:
does this work? Wondering why the existing code is written like it is
```suggestion
        match &self.err {
```

Reply:
Also occurs a couple of other times in this file